python/smells/open-never-closed.yml

```diff
@@ -0,0 +1,47 @@
+rules:
+  - id: open-never-closed
+    patterns:
+        - pattern-not-inside: |
+            $F = open(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = io.open(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = os.open(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = tarfile.open(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = ZipFile.open(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = tempfile.TemporaryFile(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = tempfile.NamedTemporaryFile(...)
+            ...
+            $F.close()
+        - pattern-not-inside: |
+            $F = tempfile.SpooledTemporaryFile(...)
+            ...
+            $F.close()
+        - pattern-either:
+            - pattern: $F = open(...)
+            - pattern: $F = io.open(...)
+            - pattern: $F = os.open(...)
+            - pattern: $F = tarfile.open(...)
```

Comment:
I think you are missing a  `pattern-not-inside` for `tarfile.open(...)` case

Reply:
Hmm, I see it above as the 4th `pattern-not-inside`